File: Planning/app/src/main/java/com/plm/planning/MainActivity.java
package com.plm.planning;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;

import android.app.Activity;
import android.os.Bundle;

import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);

        CustomView myDrawingView = new CustomView(this);
        setContentView(myDrawingView);

        try {
            Model.getTasks().add( new Task("31/08/2023","06/09/2023"));
        } catch (ParseException e) {
            throw new RuntimeException(e);
        }
    }

}

File: Couroutines1/app/src/main/java/com/plm/couroutines1/MainActivity.kt
package com.plm.couroutines1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import com.plm.couroutines1.databinding.ActivityMainBinding
import kotlinx.coroutines.*
import kotlinx.coroutines.Dispatchers.IO
import kotlin.coroutines.CoroutineContext
import kotlin.coroutines.EmptyCoroutineContext

class MainActivity : AppCompatActivity() {
    lateinit var job:Job
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        binding = ActivityMainBinding.inflate(layoutInflater)

        job= MainScope().launch {
            for(i in 0..100){
                Log.d("PABLO","hilo "+i)
            }
        }

        binding.button.setOnClickListener(
            View.OnClickListener {
                if(job.isCompleted){
                    Log.d("PABLO","HOLA!!!!!")
                }
            }
        )
    }


}